lib/engines/content_object_store/app/lib/content_object_store/publishable.rb

```diff
@@ -2,6 +2,26 @@ module ContentObjectStore
   module Publishable
     class PublishingFailureError < StandardError; end
 
+    def publish_with_rollback(schema:, document_title:, details:)
+      raise ArgumentError, "Local database changes not given" unless block_given?
+
+      ActiveRecord::Base.transaction do
+        content_block_edition = yield
```

Comment:
Here we perform either the Whitehall logic to create or update a content block, before letting the module code take care of the requests and rollbacks.